packages/core/src/events/index.ts

```diff
@@ -271,7 +271,7 @@ export class EventFactory {
 
     const evt: CoreSegmentEvent = {
       ...body,
-      messageId: this.createMessageId(),
+      messageId: (event.messageId) ? event.messageId : this.createMessageId(),
```

Comment:
event.messageId || this.createMessageId() would be more idiomatic IMO.

Reply:
@silesky , updated code base & added it into validateEvent.